src/codeCache.cpp

```diff
@@ -60,22 +51,22 @@ MethodName::MethodName(jmethodID method, const char* sep) {
         jvmti->GetMethodDeclaringClass(method, &method_class);
         jvmti->GetClassSignature(method_class, &class_name, NULL);
 
-        snprintf(_buf, sizeof(_buf), "%s%s%s", fixClassName(class_name), sep, method_name);
+        snprintf(_buf, sizeof(_buf), "%s.%s", fixClassName(class_name, dotted), method_name);
         _str = _buf;
 
         jvmti->Deallocate((unsigned char*)class_name);
         jvmti->Deallocate((unsigned char*)method_name);
     }
-
-    escape((char *)_str);
 }
 
-char* MethodName::fixClassName(char* name) {
-    char* s;
-    for (s = name + 1; *s; s++) {
-        if (*s == '/') *s = '.';
+char* MethodName::fixClassName(char* name, bool dotted) {
+    if (dotted) {
+        for (char* s = name + 1; *s; s++) {
+            if (*s == '/') *s = '.';
+        }
     }
-    s[-1] = 0;
+
+    name[strlen(name) - 1] = 0;
     return name + 1;
```

Comment:
[Class signature](https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.7.9.1) is a string of form `Ljava/lang/Thread;`
So we have to remove the first `L` and the last `;`